[SELS-TASK] User cannot take course already taken

drei001 opened the pull request:
[Commands]
```
Create a virtual environment : virtualenv "e_learning"
Install requirements: "pip install requirements.txt"
Activate virtual environment: e_learning/Scripts/Activate
change directory to backend run "python manage.py runserver"
change directory to frontend run "npm run start"

Generating fake data: "python manage.py createdata"
"createdata" command will generate fake data's depending on the queryset that has been specified on its source code, so in every commit this might vary, this is just a command used for testing.
```

[Expected Output]
![image](https://user-images.githubusercontent.com/91608901/138389223-1bd55acc-4d44-46d5-9f54-f56e302ca16d.png)
### The user can quickly see which course have already been taken
### Protected route has been implemented to protect the user going in directly to course taken link

drei001 committed:
Implemented user course validation

drei001 committed:
Bug fixes

inahsanjuan commented:
@drei001 
In this branch I could not signup, can you please confirm?
I removed all cookies/sessions, and this is after resetting db.
![image](https://user-images.githubusercontent.com/20507565/138975829-635a145d-e387-4b10-abd8-5097bc78ade9.png)

drei001 committed:
fix(axios-interceptors): exclude sending of token in signup and login

Added a conditional statement on api interceptors to exclude token when logging in or signing up

inahsanjuan commented on frontend/src/pages/Categories/CategoryList.js:
```diff
@@ -21,15 +26,36 @@ const CategoryList = () => {
+        to={`/categories/${category.id}`}
+        className="ui primary button"
+        key={category.id}
+        onClick={fetchStudentLesson(category.id)}
```
@drei001 This is calling an API request multiple times. and can cause performance problems
Better to use endpoint that can fetch all needed data once
![image](https://user-images.githubusercontent.com/20507565/138995143-6258f7ce-0a6f-4715-88d8-4777687e063b.png)

drei001 committed:
perf(Category-List,Category-Detail-and-Reducers): Prevent multiple api calls on category list

Complete remove student_lesson state in category detail useeffect dependency and refactor student reducer lessons
